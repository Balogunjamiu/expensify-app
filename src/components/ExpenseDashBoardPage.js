import React from 'react'
import ExpenseList from './expenseList'
import ExpenseListFilters from './ExpenseListFilters'


export const ExpenseDashBoardPage = () =>(
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)
export default ExpenseDashBoardPage