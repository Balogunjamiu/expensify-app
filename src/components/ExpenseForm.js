import React from 'react'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'
import 'react-dates/initialize';

const now = moment();
console.log(now.format('MMM Do, YYYY'))

export default class ExpenseForm extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            description: props.expense? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt): moment(),
            calendarFocus:true,
            error:''
            }
    }
    onDescriptionChange=(e)=>{
        const description = e.target.value
        this.setState(()=>({description}))
    }
    onNoteChange=(e)=>{
        const note = e.target.value
        this.setState(()=>({note}))
    }
    onAmountChange=(e)=>{
        const amount = e.target.value
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>({amount}))
        }
    }
    onDateChange = (createdAt)=> {
        if(createdAt){
            this.setState(()=>({createdAt}))
        }
    };
    onFocusChange = ({ focused })=>{
        this.setState(()=>({calendarFocus: focused}))
    }
    onSubmit =(e)=>{
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=>({error:'Please provide description and amount'}))
        }else{
            this.setState(()=>({error:''}))
            this.props.onSubmit({
                description: this.state.description,
                amount:parseFloat(this.state.amount, 10) * 100,
                createdAt:this.state.createdAt.valueOf(),
                note: this.state.note
            })        
        }
    };
    render(){
        return (
                <form className="form" onSubmit={this.onSubmit}>
                {this.state.error&&<p className="form__error">{this.state.error}</p>}
                    <input type="description"
                    className="text-input"
                    placeholder="description" 
                    autoFocus  value={this.state.description}
                     onChange={this.onDescriptionChange}/>
                    <input type="text"
                    className="text-input"
                    value={this.state.amount }
                    onChange={this.onAmountChange}
                    placeholder="Amount" />
                    <SingleDatePicker
                    date={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocus}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={()=> false}
                    />
                    <textarea placeholder="Add note for your expenses (optional)"
                    className="textarea"
                    value={this.state.note} onChange={this.onNoteChange}></textarea>
                    <div>
                    <button className="button">Save Expense</button>
                    </div>
                </form>
        )
    }
}