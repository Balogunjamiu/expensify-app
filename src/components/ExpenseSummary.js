import React from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import selectExpenseTotal from '../selectors/expense-total'
import selectExpenses from '../selectors/expenses'
import nonVisibleExpenses from '../selectors/non-visibleExpenses'
import numeral from 'numeral';

export const ExpenseSummary=({expenseCount, expenseTotal, unseen})=>{
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    return(
        <div className="page-header">
            <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{numeral( expenseTotal/ 100).format('$0,0.00')}</span></h1>
        <div className="page-header__action">
            <Link className="button" to="/create">Add Expense</Link>
        </div>
        {unseen.length >=1 && 
        <p>There {unseen.length <= 1 ?'is': 'are'} {unseen.length} {unseen.length ===1? 'expense': 'expenses'}  unseen, due to the filter, feel free to cancel the filter</p>
        }
            </div>
    </div>
    )
}
const mapStateToProps=(state)=>{
   const visibleExpenses = selectExpenses(state.expenses,state.filters)
   return {
       expenseCount: visibleExpenses.length,
       expenseTotal: selectExpenseTotal(visibleExpenses),
       unseen: nonVisibleExpenses(state.expenses)    
   }
}
export default connect(mapStateToProps)(ExpenseSummary)