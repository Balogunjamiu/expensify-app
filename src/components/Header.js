import React from 'react'
import { NavLink } from 'react-router-dom'
import {connect} from 'react-redux'
import {startLogout } from '../actions/auth'

 export const Header = ({startLogout}) =>(
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="is-active" >DashBoard</NavLink>
        <NavLink to="/create"activeClassName="is-active">Create</NavLink>
        <NavLink to="/help" activeClassName="is-active">help page</NavLink>
        <button onClick={startLogout}>Logout</button>
    </header>
)
const mapDispatchToProps=(dispatch)=>({
    startLogout:() => dispatch(startLogout())
})
export default connect(undefined, mapDispatchToProps)(Header)