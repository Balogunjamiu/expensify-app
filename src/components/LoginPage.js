import React from 'react'
import {connect} from 'react-redux'
import { startLoginWithGithub, startLoginWithGoogle } from '../actions/auth'
export const LoginPage = ({startLoginWithGithub, startLoginWithGoogle})=>(
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It is time to get your expenses under control</p>
            <button className="button" onClick={startLoginWithGoogle}>Login with Google</button>
            <button className="button gap" onClick={startLoginWithGithub}>Login with Github</button>
        </div>
    </div>
)
const mapDispatchToProps = (dispatch)=>({
    startLoginWithGoogle:() => dispatch(startLoginWithGoogle()),
    startLoginWithGithub:() => dispatch(startLoginWithGithub())

})
export default connect(undefined, mapDispatchToProps)(LoginPage)
