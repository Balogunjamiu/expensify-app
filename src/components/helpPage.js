import React from 'react'
const helpPage = () =>(
    <div>
        This is from my help page
    </div>
)
export default helpPage