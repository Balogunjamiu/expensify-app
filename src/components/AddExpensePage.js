import React from 'react'
import {connect} from 'react-redux'
import {startAddExpense} from '../actions/expenses'
import ExpenseForm from './ExpenseForm'


export class AddExpensePage extends React.Component{
    onSubmit = (expense)=>{
        this.props.startAddExpense(expense)
        this.props.history.push('/dashboard')
    }
    backHome=()=>{
        this.props.history.push('/dashboard')
    }
    render(){

        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                    </div>
                </div >
                <div className="content-container">
                <ExpenseForm 
                onSubmit= {this.onSubmit} />
                <button className="button" onClick={this.backHome}>Home</button>
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch)=>{
    return{
        startAddExpense:(expense)=> dispatch(startAddExpense(expense))
        
    }
    }
export default connect(undefined,mapDispatchToProps )(AddExpensePage)
