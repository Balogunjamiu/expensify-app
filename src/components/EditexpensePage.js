import React from 'react'
import {connect} from 'react-redux'
import ExpenseForm from './ExpenseForm'
import {startRemoveExpense, startEditExpense} from '../actions/expenses'

 export class EditExpensePage extends React.Component{
     onSubmit=(expense)=>{
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/dashboard')
     }
     onRemove=()=>{
        this.props.startRemoveExpense({id: this.props.expense.id})
        this.props.history.push('/dashboard')
     }
    render(){
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                <h1 className="page-header__title">Edit expense</h1>
                    </div>
                </div>
                <div className="content-container">
                <ExpenseForm 
                expense={this.props.expense}
                onSubmit={this.onSubmit}
                />
                <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>
                </div>
            </div>
        )
    }
}
const mapDispatchToProps = (dispatch, props) => {
    return{
        startEditExpense:(id,expense)=>dispatch(startEditExpense(id, expense)),
        startRemoveExpense:(data)=> dispatch(startRemoveExpense(data))
    }
}
 const mapStateToProps = (state, props)=>{
    return {
        expense: state.expenses.find((expense)=>expense.id === props.match.params.id)
    }
 }
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)