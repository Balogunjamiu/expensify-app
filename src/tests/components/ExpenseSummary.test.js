import  React from 'react'
import {shallow} from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'

test('should render Expense total page corectly with one Expense', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={234}/>)
    expect(wrapper).toMatchSnapshot();
})

test('should render Expense total page corectly with multiple Expense', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={23} expenseTotal={23412345675654}/>)
    expect(wrapper).toMatchSnapshot();
})