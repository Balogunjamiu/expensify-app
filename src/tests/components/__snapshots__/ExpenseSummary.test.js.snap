// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense total page corectly with multiple Expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $234,123,456,756.54
  </h1>
</div>
`;

exports[`should render Expense total page corectly with one Expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.34
  </h1>
</div>
`;
